airflow/contrib/operators/sagemaker_tuning_operator.py

```diff
@@ -0,0 +1,139 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow.contrib.hooks.sagemaker_hook import SageMakerHook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import AirflowException
+
+
+class SageMakerTuningOperator(BaseOperator):
+
+    """
+       Initiate a SageMaker hyper-parameter tuning job
+
+       This operator returns The ARN of the tuning job created in Amazon SageMaker
+
+       :param config:
+       The configuration necessary to start a tuning job (templated)
+       :type config: dict
+       :param region_name: The AWS region_name
+       :type region_name: str
+       :param aws_conn_id: The AWS connection ID to use.
+       :type aws_conn_id: str
+       :param wait_for_completion: if the operator should block
+       until tuning job finishes
+       :type wait_for_completion: bool
+       :param check_interval: if wait is set to be true, this is the time interval
+       in seconds which the operator will check the status of the tuning job
+       :type check_interval: int
+       :param max_ingestion_time: if wait is set to be true, the operator will fail
+       if the tuning job hasn't finish within the max_ingestion_time in seconds
+       (Caution: be careful to set this parameters because tuning can take very long)
+       :type max_ingestion_time: int
+
+       **Example**:
+           The following operator would start a tuning job when executed
+
+            sagemaker_tuning =
+               SageMakerTuningOperator(
+                   task_id='sagemaker_tuning',
+                   config=config,
+                   region='us-west-2',
+                   check_interval=20,
+                   max_ingestion_time=3600,
+                   aws_conn_id='aws_customers_conn',
+               )
+       """
+
+    template_fields = ['config', 'region_name']
+    template_ext = ()
+    ui_color = '#ededed'
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 region_name=None,
+                 aws_conn_id='sagemaker_default',
+                 wait_for_completion=True,
+                 check_interval=30,
+                 max_ingestion_time=None,
+                 *args, **kwargs):
+        super(SageMakerTuningOperator, self)\
+            .__init__(*args, **kwargs)
+
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.config = config
+        self.wait_for_completion = wait_for_completion
+        self.check_interval = check_interval
+        self.max_ingestion_time = max_ingestion_time
+
+    def evaluate(self):
```

Comment:
Should `evaluate` be implemented in a superclass and driven off a data table in conjunction with my comment below?

```python
self.integer_fields = [['HyperParameterTuningJobConfig', 'ResourceLimits', 'MaxNumberOfTrainingJobs'],
                       ['TrainingJobDefinition', 'ResourceConfig', 'InstanceCount'],
                       ['TrainingJobDefinition', 'ResourceConfig', 'VolumeSizeInGB'],
                       ['TrainingJobDefinition', 'StoppingCondition', 'MaxRuntimeInSeconds']]
```

If you wanted to be really clever, you could have a recursive structure:
```python
self.integer_fields = [['HyperParameterTuningJobConfig', 'ResourceLimits', 'MaxNumberOfTrainingJobs'],
                       ['TrainingJobDefinition', 
                           [['ResourceConfig', ['InstanceCount', 'VolumeSizeInGB']]],
                            ['StoppingCondition', 'MaxRuntimeInSeconds']]]]
```

But that's probably overkill :)

Reply:
Also, `evaluate` isn't a very descriptive name of what this function does.

Reply:
Now each sagemaker operator has its own integer_fields that can be evaluated.